newIDE/app/src/ProjectCreation/CreateProjectDialog.js

```diff
@@ -67,99 +67,189 @@ export type OnCreateFromExampleShortHeaderFunction = ({|
 
 type Props = {|
   ...CreateProjectDialogWithComponentsProps,
-  examplesComponent: any,
+  onCreateBlank: OnCreateBlankFunction,
   onCreateFromExampleShortHeader: OnCreateFromExampleShortHeaderFunction,
 |};
 
-export default class CreateProjectDialog extends React.Component<Props, State> {
-  state = {
-    currentTab: this.props.initialTab,
-    outputPath: app ? findEmptyPathInDefaultFolder(app) : '',
-  };
+const CreateProjectDialog = ({
+  open,
+  onClose,
+  onOpen,
+  onCreateFromExampleShortHeader,
+  onCreateBlank,
+  initialTab,
+}: Props) => {
+  const [currentTab, setCurrentTab] = React.useState<CreateProjectDialogTabs>(
+    initialTab
+  );
+  const [outputPath, setOutputPath] = React.useState<string>(
+    app ? findEmptyPathInDefaultFolder(app) : ''
+  );
+  const [isOpening, setIsOpening] = React.useState<boolean>(false);
+  const [newProjectName, setNewProjectName] = React.useState<string>('');
+  const [
+    selectedExampleShortHeader,
+    setSelectedExampleShortShortHeader,
+  ] = React.useState<?ExampleShortHeader>(null);
+  const [
+    preCreationDialogOpen,
+    setPreCreationDialogOpen,
+  ] = React.useState<boolean>(false);
+
+  const openPreCreationDialog = React.useCallback((open: boolean) => {
+    if (open) {
+      setOutputPath(app ? findEmptyPathInDefaultFolder(app) : '');
+      setNewProjectName(generateName());
+    }
+    setPreCreationDialogOpen(open);
+  }, []);
+
+  const actions = React.useMemo(
+    () => [
+      <RaisedButton
+        key="create-blank"
+        label={<Trans>Create a blank project</Trans>}
+        primary={false}
+        onClick={() => openPreCreationDialog(true)}
+      />,
+      <FlatButton
+        key="close"
+        label={<Trans>Close</Trans>}
+        primary={false}
+        onClick={onClose}
+      />,
+    ],
+    [onClose, openPreCreationDialog]
+  );
 
-  _onChangeTab = (newTab: CreateProjectDialogTabs) => {
-    this.setState({
-      currentTab: newTab,
-    });
+  const secondaryActions = React.useMemo(
+    () => [
```

Comment:
should we simplify this to 
```
if (currentTab === 'games-showcase') return [<FlatButton ... />]
if (currentTab === 'examples') return [<FlatButton .../>]
```

rather than hiding the logic inside the array

Reply:
Oh yes you're right, I brainless copy-pasted